.github/workflows/codespell.yml

```diff
@@ -0,0 +1,19 @@
+---
+name: Codespell
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
```

Comment:
Why do we need a codespell job as well as having it in pre-commit? Surely one is sufficient?

Reply:
I have prepared this PR without having pre-commit enabled. People can contribute PRs similarly and nobody would know that they have not used pre-commit. So this is just an extra quick check.

Reply:
We do run pre-commit in our CI, so an additional job should not be necessary.

Reply:
ah, ok, removed workflow (and rebased).  Also pushed a test reversal of typo fixes to see if indeed gets detected/failing

Reply:
cool -- it worked and failed in https://results.pre-commit.ci/run/github/1446467/1679924128.Zy5acL8XSw60JY_n78_gjw , I hard reset before that commit.